src/zenml/zen_server/auth.py

```diff
@@ -533,7 +544,28 @@ def authenticate_external_user(external_access_token: str) -> AuthContext:
 
     if 200 <= auth_response.status_code < 300:
         try:
-            payload = auth_response.json()
+            content_type = (
+                auth_response.headers["Content-Type"].split("/").pop()
+            )
+
+            # A jwt token is returned when the content type is application/jwt and is decoded differently than a token with am application/json content type
+            if content_type == "jwt":
+                authorization_token = auth_response.text
+
+                # OpenID Connect 5.3.2. Successful UserInfo Response states that the UserInfo response MAY be encrypted without also being signed
+                decoded_token = jwt.decode(
```

Comment:
This is the line of code that concerns me the most. The decoded JWT token is parsed into a `ExternalUserModel` further down the line and it will fail authentication if it doesn't contain the necessary `id` UUID and `email` fields. Have you tested that this works with an actual middleware ?